iXBRLViewerPlugin/viewer/src/js/interval.js

```diff
@@ -0,0 +1,62 @@
+// See COPYRIGHT.md for copyright information
+
+import Decimal from 'decimal.js';
+
+export class Interval {
+
+    constructor(a, b) {
+        this.a = typeof a == 'object' ? a : new Decimal(a);
+        this.b = typeof b == 'object' ? b : new Decimal(b);
+    }
+
+    static fromFact(fact) {
+        if (!fact.isNumeric()) {
+            return undefined;
+        }
+        const decimals = fact.decimals();
+        let width = 0;
+        if (decimals !== undefined) {
+            const x = new Decimal(10);
+            width = x.pow(0-decimals).times(0.5);
+        }
+        try {
+            const value = new Decimal(fact.value());
+            return new Interval(value.minus(width), value.plus(width));
+        } catch (e) {
+            if (e instanceof Error && /DecimalError/.test(e.message)) {
```

Comment:
Do `minus`/`plus` somehow throw (if so, how?), or does this only apply to the ctor?  If only the ctor, would you mind refactoring so the `try`/`catch` only wraps the `new Decimal` to make that more obvious?

Reply:
I'm pretty sure it's just the `new Decimal()` constructor, and I assume it's only there to deal with values that would be considered schema invalid anyway.  Refactored in 05b811a